modin/pandas/dataframe.py

```diff
@@ -819,12 +819,57 @@ def explode(
             pandas.DataFrame.explode, column, ignore_index=ignore_index
         )
 
+    def _update_var_dicts_in_kwargs(self, expr, **kwargs):
+        """
+        Copy variables with "@" prefix in `local_dict` and `global_dict`.
+
+        Parameters
+        ----------
+        expr : str
+            The expression string to search variables with "@" prefix.
+        **kwargs : dict
+            See the documentation for eval() for complete details on the keyword arguments accepted by query().
+
+        Returns
+        -------
+        kwargs : dict
+            Dict with updated `local_dict` and `global_dict` keys.
+        """
+        import re
+
+        frame = sys._getframe()
```

Comment:
```suggestion
        if "@" not in expr:
            return kwargs
        frame = sys._getframe()
```
let's insert some quick check for a hot path (I would expect that _most_ queries do not reference local variables).